File: python/2410.maximum-matching-of-players-with-trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: list[int], trainers: list[int]) -> int:
        m, n = len(players), len(trainers)

        i, j = 0, 0
        count = 0

        players.sort()
        trainers.sort()

        while i < m and j < n:
            if players[i] <= trainers[j]:
                i += 1
                j += 1
                count += 1
            else:
                j += 1

        return count

File: cpp/2410.maximum-matching-of-players-with-trainers.cpp
#include <vector>
using std::vector;

class Solution {
public:
  int matchPlayersAndTrainers(vector<int> &players, vector<int> &trainers) {
    const int m = players.size(), n = trainers.size();
    int count = 0;

    std::sort(players.begin(), players.end());
    std::sort(trainers.begin(), trainers.end());

    for (int i = 0, j = 0; i < m && j < n; j++) {
      if (players[i] <= trainers[j]) {
        count++;
        i++;
      }
    }

    return count;
  }
};